chapters/en/slides/module2/module2_04.Rmd

```diff
@@ -0,0 +1,143 @@
+---
+params:
+  dynamictitle: "module2_04"
+title: "`r params$dynamictitle`"
+output:
+  md_document:
+    variant: gfm
+    pandoc_args: "--markdown-headings=atx"
+---
+
+```{r setup, include=FALSE}
+source('../../../../slide-setup.R')
+# Remember to also manually update the YAML title above
+```
+
+type: slides
+
+# Creating subplots via faceting
+
+Notes:
+Exploring mutiple vars
+
+---
+
+```{python}
+import pandas as pd
+import altair as alt
+
+url = 'https://raw.githubusercontent.com/UofTCoders/workshops-dc-py/master/data/processed/world-data-gapminder.csv'
+gm = pd.read_csv(url, parse_dates=['year'])
+gm2018 = gm[gm['year'] == '2018']
+
+alt.Chart(gm2018).mark_bar().encode(
+    x=alt.X('life_expectancy', bin=alt.Bin(maxbins=30)),
+    y='count()',
+    color='region')
+```
+
+Notes:
+We have previously seen how we can explore categorical dataframe columns
+by encoding them as the color channel of our plots.
+Doing this with barplots allows us to drill down into two categorical columns
+at the same time:
+the one on the y-axis
+the one represented by color.
+
+Altair creates a stacked bar chart by default.
+Just like with the stacked area chart,
+this is good when the total height of each bar
+is the most important,
+but it is difficult to compare
+the length of the different colored segments against each other,
+both within a bar and between bars.
+
+---
+
+## Faceting creates subplots for different subsets of the data
+
+```{python}
+alt.Chart(gm2018).mark_bar().encode(
+    x=alt.X('life_expectancy', bin=alt.Bin(maxbins=30)),
+    y='count()',
+    color='region'
+).properties(width=200, height=200
+).facet('region')
+```
```

Comment:
I know they might not fit the slide, but wouldn't these be better to compare if we stacked them vertically? If it's super difficult on the slide, or looks funny, we might want to mention it at least, and then show that on another slide, where we just focus on that. That seems like missing material a bit... When to stack vertically, horizontally, or in a grid...

Reply:
I don't have any directly good guideline for when to stack in different ways, this would be interesting to discuss! I also have found a way to modify slight height, not sure if it will work for an indivisual slide but could try. I will ping you when  I revisit this.